core/src/main/java/com/datastrato/gravitino/storage/relational/session/SqlSessionFactoryHelper.java

```diff
@@ -0,0 +1,119 @@
+ * should be initialized only once.
+ */
+public class SqlSessionFactoryHelper {
+  private static volatile SqlSessionFactory sqlSessionFactory;
+  private static final SqlSessionFactoryHelper INSTANCE = new SqlSessionFactoryHelper();
+
+  public static SqlSessionFactoryHelper getInstance() {
+    return INSTANCE;
+  }
+
+  private SqlSessionFactoryHelper() {}
+
+  @VisibleForTesting
+  static void setSqlSessionFactory(SqlSessionFactory sessionFactory) {
+    sqlSessionFactory = sessionFactory;
+  }
+
+  /**
+   * Initialize the SqlSessionFactory object.
+   *
+   * @param config Config object to get the jdbc connection details from the config.
+   */
+  @SuppressWarnings("deprecation")
+  public void init(Config config) {
+    // Initialize the data source
+    BasicDataSource dataSource = new BasicDataSource();
+    dataSource.setUrl(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_URL));
+    dataSource.setDriverClassName(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_DRIVER));
+    String dbType = config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_DB_TYPE);
+    if (dbType.equals(Configs.DEFAULT_ENTITY_RELATIONAL_JDBC_BACKEND_DB_TYPE)) {
+      dataSource.setUsername(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_USER));
+      dataSource.setPassword(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_PASSWORD));
+    }
+    // Close the auto commit, so that we can control the transaction manual commit
+    dataSource.setDefaultAutoCommit(false);
+    dataSource.setMaxWaitMillis(1000L);
+    dataSource.setMaxTotal(20);
+    dataSource.setMaxIdle(5);
+    dataSource.setMinIdle(0);
+    dataSource.setLogAbandoned(true);
+    dataSource.setRemoveAbandonedOnBorrow(true);
+    dataSource.setRemoveAbandonedTimeout(60);
+    dataSource.setTimeBetweenEvictionRunsMillis(Duration.ofMillis(10 * 60 * 1000L).toMillis());
+    dataSource.setTestOnBorrow(BaseObjectPoolConfig.DEFAULT_TEST_ON_BORROW);
+    dataSource.setTestWhileIdle(BaseObjectPoolConfig.DEFAULT_TEST_WHILE_IDLE);
+    dataSource.setMinEvictableIdleTimeMillis(1000);
+    dataSource.setNumTestsPerEvictionRun(BaseObjectPoolConfig.DEFAULT_NUM_TESTS_PER_EVICTION_RUN);
+    dataSource.setTestOnReturn(BaseObjectPoolConfig.DEFAULT_TEST_ON_RETURN);
+    dataSource.setSoftMinEvictableIdleTimeMillis(
+        BaseObjectPoolConfig.DEFAULT_SOFT_MIN_EVICTABLE_IDLE_TIME.toMillis());
+    dataSource.setLifo(BaseObjectPoolConfig.DEFAULT_LIFO);
+
+    // Create the transaction factory and env
+    TransactionFactory transactionFactory = new JdbcTransactionFactory();
+    Environment environment = new Environment("development", transactionFactory, dataSource);
+
+    // Initialize the configuration
+    Configuration configuration = new Configuration(environment);
+    configuration.addMapper(MetalakeMetaMapper.class);
+
+    // Create the SqlSessionFactory object, it is a singleton object
+    if (sqlSessionFactory == null) {
```

Comment:
Maybe this will be checked similar issue by static code checker.
https://github.com/datastrato/gravitino/issues/1935

Reply:
Is there any way for me to check with the static code checker you mentioned?

Reply:
cc @justinmclean Could you provide a brief way?

Reply:
One simple way is to install SonarLint in your IDE.

Reply:
I used `SonarLint` to check the file, and no volatile-related prompts appeared. It only checked the use of singletons and paid attention to the usage prompts.
![image](https://github.com/datastrato/gravitino/assets/26177232/41617792-e954-4292-89d6-187f963cf031)

Reply:
ok.

Reply:
By the way, some configurations of `SqlSessionFactory` comes from the `Config`, and it is unlikely to use static code blocks or static variables to complete the initialization of this singleton.